From cf8ad3668817519d312251a6afceb7c36e73e22d Mon Sep 17 00:00:00 2001
From: Lukas Elsner <open@mindrunner.de>
Date: Fri, 18 May 2012 01:59:51 +0200
Subject: [PATCH] ..

---
 Makefile.am |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 5ee6b85..95acb88 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -13,7 +13,5 @@ endif
 
 if RUN_LDCONFIG
 install-exec-hook:
-	@printf "Running ldconfig to ensure libraries installed in system library directories are added to the dynamic linker cache ... "
-	@-@NETA_LDCONFIG@
-	@printf "done\n"
+	@printf "Not running ldconfig to ensure libraries installed in system library directories are added to the dynamic linker cache, because it sucks!"
 endif
-- 
1.7.8.6

