--- a/src/fl_sources/google_source_feed.c	2009-05-02 04:53:16.000000000 +1000
+++ b/src/fl_sources/google_source_feed.c	2010-02-24 18:56:06.249297068 +1100
@@ -343,7 +343,7 @@
 
 	if (!g_str_equal (request->source, GOOGLE_READER_BROADCAST_FRIENDS_URL)) { 
 		gchar* source_escaped = g_uri_escape_string(request->source, NULL, TRUE);
-		gchar* newUrl = g_strdup_printf ("http://www.google.com/reader/atom/feed/%s", source_escaped);
+		gchar* newUrl = g_strdup_printf ("http://www.google.com/reader/atom/feed/%s?n=1000", source_escaped);
 		update_request_set_source (request, newUrl);
 		g_free (newUrl);
 		g_free (source_escaped);
