--- a/src/psb_dga.c
+++ b/src/psb_dga.c
@@ -298,7 +298,7 @@
 PSB_Sync(ScrnInfoPtr pScrn)
 {
     ScreenPtr pScreen = pScrn->pScreen;
-    WindowPtr pRoot = WindowTable[pScreen->myNum];
+    WindowPtr pRoot = pScreen->root;
     char buffer[4];
 
     pScreen->GetImage(&pRoot->drawable, 0, 0, 1, 1, ZPixmap, ~0L, buffer);
@@ -310,13 +310,13 @@
 {
     GCPtr pGC;
     DrawablePtr pDrawable;
-    XID vals[1];
+    ChangeGCVal vals[1];
     xRectangle r;
 
     if (!xf86_dga_get_drawable_and_gc(pScrn, &pDrawable, &pGC))
 	return;
-    vals[0] = color;
-    ChangeGC(pGC, GCForeground, vals);
+    vals[0].val = color;
+    ChangeGC(NullClient, pGC, GCForeground, vals);
     ValidateGC(pDrawable, pGC);
     r.x = x;
     r.y = y;
