=== modified file 'psb-kernel-source/psb_drv.c'
--- psb-kernel-source/psb_drv.c	2010-10-26 14:49:29 +0000
+++ psb-kernel-source/psb_drv.c	2010-12-04 22:33:15 +0000
@@ -1022,6 +1022,9 @@
 	drm_exit(&driver);
 }
 
+/* removed in Xserver 1.7, add it again so that proprietary Xpsb can be loaded */
+void xf86AddModuleInfo(pointer info, pointer module){ }
+
 module_init(psb_init);
 module_exit(psb_exit);
 

